idl/subscribe.x

```diff
@@ -0,0 +1,131 @@
+namespace mazzaroth
+{
+  struct ReceiptSubscription
+  {
+    TransactionFilter transactionFilter;
+ 
+    ReceiptFilter receiptFilter;
+  };
+
+  struct ReceiptSubscriptionResult
+  {
+    Receipt receipt;
+
+    ID transactionID;
+  };
+
+  enum ValueFilterType
+  {
+    NONE = 0,
+    STRING = 1,
+    HASH32 = 2,
+    HASH64 = 3,
+    UHYPER = 4,
+    BOOL = 5,
+  };
+
+  union ValueFilter switch (ValueFilterType Type) 
+  {
+    case NONE:
+      void;
+    case STRING:
+      string regex<256>;
+    case HASH32:
+      Hash hash32Value;
+    case HASH64:
```

Comment:
Could this ValueFilterType be called SIGNATURE?

Reply:
It could, then it would be inconsistent with Hash32 type. I don't like that idea.

I can make new typedefs for these if you have a problem with the naming. We would then have ID, Hash and Hash32 that are 32 byte hashes with Signature and Hash64 as 64 byte hashes. This would also mean new utility functions for translating between byteslices and these new typedefs.

Reply:
That is fair.  I don't think it makes sense to add additional typedefs that are not needed.

Reply:
After we talked about it, I've added specific types rather than reusing Hash and Signature. We could follow up with cleanup at some point in the future for our various opaque typdefs.

Reply:
Thanks!